pkg/commands/componentarchive/create_test.go

```diff
@@ -0,0 +1,88 @@
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	cdv2 "github.com/gardener/component-spec/bindings-go/apis/v2"
+	"github.com/gardener/component-spec/bindings-go/codec"
+	"github.com/gardener/component-spec/bindings-go/ctf"
+
+	"github.com/gardener/component-cli/pkg/commands/componentarchive"
+)
+
+var _ = Describe("Create", func() {
+
+	var testdataFs vfs.FileSystem
+
+	BeforeEach(func() {
+		baseFs, err := projectionfs.New(osfs.New(), "./testdata")
+		Expect(err).ToNot(HaveOccurred())
+		testdataFs = layerfs.New(memoryfs.New(), baseFs)
+	})
+
+	It("should create a component archive and overwrite with a newer version", func() {
+		opts := &componentarchive.CreateOptions{}
+		opts.Name = "example.com/component/name"
+		opts.Version = "v0.0.1"
+		opts.ComponentArchivePath = "./create-test"
+		//opts.ComponentNameMapping = "urlPath"
+		opts.BaseUrl = "example.com/testurl"
+
+		err := testdataFs.Mkdir(opts.ComponentArchivePath, os.ModePerm)
+		Expect(err).ToNot(HaveOccurred())
+
+		Expect(opts.Run(context.TODO(), logr.Discard(), testdataFs)).To(Succeed())
+
+		data, err := vfs.ReadFile(testdataFs, filepath.Join(opts.ComponentArchivePath, ctf.ComponentDescriptorFileName))
+		Expect(err).ToNot(HaveOccurred())
+
+		cd := &cdv2.ComponentDescriptor{}
+		Expect(codec.Decode(data, cd)).To(Succeed())
+		Expect(cd.Name).To(Equal(opts.Name), "component name should be the same")
+		Expect(cd.Version).To(Equal(opts.Version), "component version should be the same")
+
+		Expect(len(cd.RepositoryContexts) > 0).To(BeTrue(), "The repository contexts should return some data")
+		repoCtx := cd.RepositoryContexts[0]
+		Expect(repoCtx.GetType()).To(Equal(cdv2.OCIRegistryType), "check the repository context")
+		ociRepoCtx := &cdv2.OCIRegistryRepository{}
+		Expect(repoCtx.DecodeInto(ociRepoCtx)).To(Succeed())
+		Expect(ociRepoCtx.BaseURL).To(Equal(opts.BaseUrl))
+
+		// check overwrite
+		opts.Version = "v0.0.2"
+		Expect(opts.Run(context.TODO(), logr.Discard(), testdataFs)).To(HaveOccurred(), "Should not overwrite existing component")
+
+		opts.Overwrite = true
+		Expect(opts.Run(context.TODO(), logr.Discard(), testdataFs)).To(Succeed(), "Should overwrite existing component")
+
+		data, err = vfs.ReadFile(testdataFs, filepath.Join(opts.ComponentArchivePath, ctf.ComponentDescriptorFileName))
+		Expect(err).ToNot(HaveOccurred())
+
+		cd = &cdv2.ComponentDescriptor{}
+		Expect(codec.Decode(data, cd)).To(Succeed())
+		Expect(cd.Name).To(Equal(opts.Name), "component name should be the same")
+		Expect(cd.Version).To(Equal(opts.Version), "component version should be the same")
+
+		Expect(len(cd.RepositoryContexts) > 0).To(BeTrue(), "The repository contexts should return some data")
+		repoCtx = cd.RepositoryContexts[0]
+		Expect(repoCtx.GetType()).To(Equal(cdv2.OCIRegistryType), "check the repository context")
+		Expect(repoCtx.DecodeInto(ociRepoCtx)).To(Succeed())
+		Expect(ociRepoCtx.BaseURL).To(Equal(opts.BaseUrl))
```

Comment:
can we have that in separate test so that we test one functionality per feature which is easier to debug in case on issues

Reply:
all done